src/tree_data.c

```diff
@@ -2788,6 +2864,13 @@ lyd_compare_single(const struct lyd_node *node1, const struct lyd_node *node2, u
     struct lyd_node_opaq *opaq1, *opaq2;
     size_t len1, len2;
 
+/* Internal flag for lyd_compare_single function.
+ * This optimization flag will be set in the options variable
+ * if the lyd_compare_parental_schemas function is called.
+ * As a result, the lyd_compare_parental_schemas is performed only once.
+ */
+#define LYD_PARENTAL_SCHEMAS_COMPARED 0x80000000
```

Comment:
I do not like mixing API and internal flags in one variable but I suppose I could let this one slide.

Reply:
So no changes are required?

Reply:
I guess, I will let @rkrejci decide.